homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -123,6 +125,7 @@ def extra_state_attributes(self):
             ATTR_TARGET: self._opendata.to_name,
             ATTR_REMAINING_TIME: f"{self._remaining_time}",
             ATTR_DELAY: self._opendata.connections[0]["delay"],
+            ATTR_POLLING_COUNT: len(self._polls)
```

Comment:
The user should normally not need to worry about this. It's the responsibility of the integration to set a default update interval that is lower than the allowed polling interval.

All integrations that support config entries support user automating the update with the entity service `homeassistant.update_entity`. If the user automates their own update interval they won't need to know this counted value.

It's a better solution to migrate this integration to config entries and using an update coordinator.

Reply:
The problem is that this sensor can be added multiple times (for different itineraries) and the polling frequency depends not only on the "update_interval" but also on the frequency of the trains on the itinerary.
When the sensor get updated it will not poll again the server if the next train is not arrived yet.
Obfuscation is sometime good but I had to look inside the source code to understand that point and be able to calculate the number of sensor (itinerary) I could use with this integration.
Again, the goal is not to restrict the frequency of polling but to allow the user to mesure how many polling are done, and how many train itinerary can exist in parallel.

Reply:
Do you mean that in the end, the config should be looking something like that : 
```yaml
swiss_public_transport:
  max_api_call_per_hour: 50 
  itineraries:
    - name : "A to B"
       from: 12345
       to : 54321
    - name : "B to A"
       from: 54321
       to : 12345
    - name : "A to C"
       from: 12345
       to : 67890
```
I mean, that could be nice, but that will make big breaking changes.

Reply:
With config entries there will be no YAML config, but design wise yes, each config entry would probably represent one item in the `itineraries` list.

We would import each platform config to a config entry to avoid making breaking changes as far as possible.

Reply:
That's sounds like the ideal solution, however I do not have yet the knowledge to make a refactoring that huge. 
Nevertheless, if an expert is keen to do it with me, that's doable. Also, I think that this PR is a good compromise until it's done. (given that the changes are light and give more information to end user I believe it's valuable)

Reply:
There are a few PR that you can use as sample for refactoring to use config-flow:
https://github.com/home-assistant/core/pulls?q=is%3Apr+is%3Aopen+label%3Aconfig-flow